platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r89/ProblemProgressEventCrossVersionTest.groovy

```diff
@@ -120,9 +121,12 @@ class ProblemProgressEventCrossVersionTest extends ToolingApiSpecification {
         verifyAll(problems[0]) {
             details?.details == expectedDetails
             definition.documentationLink?.url == expectedDocumentation
-            locations.size() == 2
+            locations.size() == targetVersion >= GradleVersion.version("8.12") ? 3 : 2
```

Comment:
🤔 I still think we should drop looking at the reported location sizes and assert the existence of the locations in a problem. This approach makes maintenance much heavier as we add more and more locations.

Reply:
Agreed. For starters, I've relaxed the assertion here.